widget_driver/doc/drivable_widgets.md

```diff
@@ -0,0 +1,79 @@
+# DrivableWidgets
+
+These are your new go-to-widgets!  
+As the name suggests, a `DrivableWidget` is a widget that can be driven by a `Driver`.
+
+---
+
+## Core concepts
+
+`DrivableWidgets` and `WidgetDrivers` are close friends. They always come in pairs. Whenever you create a `DrivableWidget`, then you will also create a `WidgetDriver`.
+
+### The driver property
+
+Each `DrivableWidget` has access to a property called `driver`. This is the instance of your `WidgetDriver`. When you want to grab some data from the driver and use it in your build method then all you need to do is to access the `driver`. Like this:
+
+```dart
+MyCoolWidget extends DrivableWidget<MyCoolWidgetDriver> {
+    Widget build(BuildContext context) {
+        return Column(children: [
+            Text(driver.coolTitle),
+            Text(driver.coolSubtitle),
+        ]);
+    }
+}
+```
+
+The `driver` is the place you go to to get data.  
+Don't grab data directly in the widget from some other data source like a `BLoC` or `Provider` or something.
+
+If there is data which your widget needs from some other source, then put the logic which grabs that data in the `driver`. The `driver` has access to the BuildContext and can easily resolve that data. And then the widget can access it via the `driver`.
+
+The important part why you need to **always** go via the `driver` when you grab data from other sources, is so that you can easily test your widget!
+
+If you always use the `driver` as a data provider in your widget, then when you test your widget, that data will be provided via the `TestDriver` and you do not need to mock it!
+
+If you grab data from another source in your widget, then the widgetTest will crash unless you provide specific mock values for that data.
+
+### Creating a DrivableWidget
+
+So how do you create a `DrivableWidget`? Well that is pretty easy. Just extend `DrivableWidget` and pass in your `WidgetDriver` type as the generic type to it. Like so:
+
+```dart
+MyCoolWidget extends DrivableWidget<MyCoolWidgetDriver> {}
+```
+
+Pro tip: Since you need the type of your `WidgetDriver` as input to your `DrivableWidget`, it is generally a good idea to first create your `Driver`.
```

Comment:
```suggestion
Pro tip: Since you need the type of your `WidgetDriver` as input to your `DrivableWidget`,
it is generally a good idea to create your `Driver` first.
```